### ISSUE TITLE
Creating password exception

### ISSUE THREAD

**Issue by tylorr** [mbos/Pass4Win#25]
This happens when I try to create my first password.

Before running this command my terminal was showing this: `~/pass (master #)` which means that it is a proper git repo. After running this command it now shows: `~/pass ((ref: re...))` and I have no idea what that means. When I try to run `git status` it tells me `fatal: Not a git repository`.

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
LibGit2Sharp.RepositoryNotFoundException: Path 'C:\Users\tylor\pass' doesn't point at a valid Git repository or workdir.
   at LibGit2Sharp.Core.Proxy.git_repository_open(String path)
   at LibGit2Sharp.Repository..ctor(String path, RepositoryOptions options)
   at Pass4Win.frmMain.btnAdd_Click(Object sender, EventArgs e)
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
Pass4Win
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/Program%20Files%20(x86)/Pass4Win/Pass4Win.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34251 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34238 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_32/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34234 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
LibGit2Sharp
    Assembly Version: 0.21.0.176
    Win32 Version: 0.21.0.176
    CodeBase: file:///C:/Program%20Files%20(x86)/Pass4Win/LibGit2Sharp.DLL
----------------------------------------
GpgApi
    Assembly Version: 1.5.0.0
    Win32 Version: 1.5.0.0
    CodeBase: file:///C:/Program%20Files%20(x86)/Pass4Win/GpgApi.DLL
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Security
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34252 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Security/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Security.dll
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

**Comment by mbos** [comment]
That's weird....
Can you reproduce it and would you be able to attach a copy of the repository (i'm assuming it's empty)?

The ref: re.. thing is that your Git Repo is damaged you can check with: `git reflog --all` if the are orphans and with `git fsck` you can check for errors in the repo

**Comment by tylorr** [comment]
After first setup point `~/pass` contains a `.git` folder and a `.gpg-id` which is an untracked file.
1. I put a password into the input and click add
2. Enter `foo` for the name
3. I get the exception.

An empty `foo.gpg` file has been added to the `~./pass` folder.

A link to download the pass folder: https://www.dropbox.com/s/xydydzie1xey2k3/pass.zip?dl=0

I can't run either `git reflog --all` or `git fsck`, I get this messsage: `fatal: Not a git repository (or any of the parent directories): .git`

**Comment by mbos** [comment]
Right thanks. So basically this uncovered two issues:
1. .gpg-id is not tracked by git **fixed**
2. The git repo gets corrupted because there will be a pull from remote, when there is no remote present. Will do tomorrow or another time.

Thanks for your feedback

**Comment by mbos** [comment]
Fixed in fe5969eb70bfdbe17ef68b6b629b818cd707cc3c
Also found an issue when creating a Repo, opened an issue with https://github.com/libgit2/libgit2sharp/issues/1158
